src/inmanta/agent/handler.py

```diff
@@ -952,6 +952,14 @@ def delete_resource(self, ctx: HandlerContext, resource: TPurgeableResource) ->
     def update_resource(self, ctx: HandlerContext, changes: Dict[str, Dict[str, Any]], resource: TPurgeableResource) -> None:
         pass
 
+    def calculate_diff(
+        self, ctx: HandlerContext, current: TPurgeableResource, desired: TPurgeableResource
+    ) -> typing.Dict[str, typing.Dict[str, typing.Any]]:
+        super().calculate_diff(ctx, current, desired)
```

Comment:
Could just use `typing.overload` as @wouterdb proposed. But I don't think this one additional super call should have any real effect on performance in practice.